### PR TITLE
fix(replay): Mask SVGs from `react-native-svg` when `maskAllVectors=true`

### PR DESCRIPTION
**Pull request by krystofwoldrich**
## :loudspeaker: Type of change

- [x] Bugfix

## :scroll: Description

This PR fixes visible SVG from `react-native-svg` the most popular library for handling SVGs in RN. RN doesn't have a built in support for SVGs.

## 🛑 Blocked by

- https://github.com/getsentry/sentry-java/pull/3546

## :bulb: Motivation and Context

Web hides SVGs by default when masking all media.

## :green_heart: How did you test it?
rn sample app

## :pencil: Checklist

- [x] I reviewed submitted code
- [ ] I added tests to verify changes
- [x] No new PII added or SDK only sends newly added PII if `sendDefaultPII` is enabled
- [x] All tests passing
- [x] No breaking changes

### CODE REVIEW

File: android/src/main/java/io/sentry/react/RNSentryModuleImpl.java [hunk 1]
```diff
@@ -326,6 +328,11 @@ private SentryReplayOptions getReplayOptions(@NotNull ReadableMap rnOptions) {
         androidReplayOptions.setRedactAllText(!rnMobileReplayOptions.hasKey("maskAllText") || rnMobileReplayOptions.getBoolean("maskAllText"));
         androidReplayOptions.setRedactAllImages(!rnMobileReplayOptions.hasKey("maskAllImages") || rnMobileReplayOptions.getBoolean("maskAllImages"));
 
+        final boolean redactVectors = !rnMobileReplayOptions.hasKey("maskAllVectors") || rnMobileReplayOptions.getBoolean("maskAllVectors");
+        if (redactVectors) {
+            androidReplayOptions.addClassToRedact("com.horcrux.svg.SvgView"); // react-native-svg
```
[the remark this answers is not in the document]
Comment (krystofwoldrich) [reply android/src/main/java/io/sentry/react/RNSentryModuleImpl.java:333]: Do we already exclude some classes from proguard/r8 in SAGP?

[the remark this answers is not in the document]
Comment (romtsn) [reply android/src/main/java/io/sentry/react/RNSentryModuleImpl.java:333]: not in SAGP, but in sentry-java, e.g. https://github.com/getsentry/sentry-java/blob/main/sentry-android-core/proguard-rules.pro

so in your case it'd be `-keepnames com.horcrux.svg.SvgView` and then you need to package the rules with the android part: https://github.com/getsentry/sentry-java/blob/25f1ca4e1636a801c17c1662f0145f888550bce8/sentry-android-core/build.gradle.kts#L31-L33

but I'm not sure if it's gonna work for react-native, but I guess it is. You probably can just add `consumerProguardFiles` to the `defaultConfig` actually

[the remark this answers is not in the document]
Comment (krystofwoldrich) [reply android/src/main/java/io/sentry/react/RNSentryModuleImpl.java:333]: The library (react-native-svg) which uses `com.horcrux.svg.*` already adds the rule.

https://github.com/software-mansion/react-native-svg/blob/main/android/proguard-rules.pro

We still might need to add it to be safe and not depending on their setup...

[the remark this answers is not in the document]
Comment (romtsn) [reply android/src/main/java/io/sentry/react/RNSentryModuleImpl.java:333]: ah ok, then we're safe I guess

**Review by brustolin — APPROVED** [review]
iOS part looks good.

**Pull request merged**